rllib/rl_trainers/torch_rl_distributed_trainer.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import ray
+import torch
+import ray.train as train
+from ray.air import session
+
+from ray.air.config import ScalingConfig
+
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+from ray.train._internal.utils import construct_train_func
+from ray.train._internal.backend_executor import BackendExecutor
+from ray.train.torch import TorchConfig
+from ray.train import TrainingIterator
+from ray.train._internal.checkpoint import CheckpointManager
+from ray.train._internal.dataset_spec import RayDatasetSpec
+from ray.util.queue import Queue
+
+
+class TorchRLDataParallelTrainer:
+
+    def __init__(self, config):
+        num_workers = config.get("num_gpus", 0)
+        use_gpu = bool(num_workers)
+        scaling_config = ScalingConfig(num_workers=num_workers, use_gpu=use_gpu)
+        queues = [Queue(maxsize=1) for _ in range(num_workers)]
+        backend_config = TorchConfig()
+        config = {"queues": queues, "rl_module_config": config["rl_module_config"], 
+            "rl_module_class": config["rl_module_class"]}
+        train_loop_per_worker = construct_train_func(
+            self.training_func,
+            config,
+            fn_arg_name="train_loop_per_worker",
+            discard_returns=True,
+        )
+        self.debug_mode = config.get("debug_mode", False)
+
+        backend_executor = BackendExecutor(
+                backend_config=backend_config,
+                num_workers=scaling_config.num_workers,
+                num_cpus_per_worker=scaling_config.num_cpus_per_worker,
+                num_gpus_per_worker=scaling_config.num_gpus_per_worker,
+                max_retries=0,
+            )
+
+        checkpoint_manager = CheckpointManager(
+                    checkpoint_strategy=None, run_dir=None
+                )
+
+        # Start the remote actors.
+        backend_executor.start(initialization_hook=None)
+
+        empty_spec = RayDatasetSpec(dataset_or_dict=None)
+
+        self.training_iterator = TrainingIterator(
+            backend_executor=backend_executor,
+            backend_config=backend_config,
+            train_func=train_loop_per_worker,
+            dataset_spec=empty_spec,
+            checkpoint_manager=checkpoint_manager,
+            checkpoint=None,
+            checkpoint_strategy=None,
+        )
+        self.curr_weights = None
+
+    @classmethod
+    def training_func(self, config):
+        if "rl_module_config" not in config:
+            raise ValueError("rl_module_config not in config")
+        rl_module_config = config["rl_module_config"]
+        rl_module_class = config["rl_module_class"]
+        rl_module = rl_module_class(rl_module_config)
+        queue = config["queues"][session.get_local_rank()]
```

Comment:
Stupid question: Who will take care of splitting up the data that arrives (as chunks) in these queues?